src/nvim/option.c

```diff
@@ -1442,15 +1442,18 @@ do_set (
               goto skip;
             }
 
-            if (adding)
-              value = *(long *)varp + value;
-            if (prepending)
-              value = *(long *)varp * value;
-            if (removing)
-              value = *(long *)varp - value;
-            errmsg = set_num_option(opt_idx, varp, value,
-                errbuf, sizeof(errbuf), opt_flags);
-          } else if (opt_idx >= 0) {                      /* string */
+            if (adding) {
+              value = *(intmax_t *)varp + value;
```

Comment:
You decide, but the issue is not aesthetics but semantics (`value` is meant to range over all valid option values). Alternatively, can't we just remove the `(intmax_t)` casts as `intmax_t` is guaranteed to be at least as big as `ptrdiff_t` ?  (the final `(NumOpt)` cast is still needed of course)